Runtimes/Supplemental/cmake/modules/FindSwiftCore.cmake

```diff
@@ -52,112 +52,102 @@ endif()
 
 include(FindPackageHandleStandardArgs)
 
+if(NOT DEFINED SwiftCore_USE_STATIC_LIBS)
+  set(SwiftCore_USE_STATIC_LIBS OFF)
+  if(NOT BUILD_SHARED_LIBS AND NOT APPLE)
+    set(SwiftCore_USE_STATIC_LIBS ON)
+  endif()
+endif()
+
 if(APPLE)
+  list(APPEND SwiftCore_INCLUDE_DIR_HINTS
+    "${CMAKE_OSX_SYSROOT}/usr/lib/swift")
+  list(APPEND SwiftCore_LIBRARY_HINTS
+    "${CMAKE_OSX_SYSROOT}/usr/lib/swift")
   # When building for Apple platforms, SwiftCore always comes from within the
   # SDK as a tbd for a shared library in the shared cache.
-  find_path(SwiftCore_INCLUDE_DIR
-      "Swift.swiftmodule"
-    HINTS
-      "${CMAKE_OSX_SYSROOT}/usr/lib/swift")
-  find_library(SwiftCore_IMPLIB
-    NAMES "libswiftCore.tbd"
-    HINTS
-      "${CMAKE_OSX_SYSROOT}/usr/lib/swift")
-  add_library(swiftCore SHARED IMPORTED GLOBAL)
-  set_target_properties(swiftCore PROPERTIES
-    IMPORTED_IMPLIB "${SwiftCore_IMPLIB}"
-    INTERFACE_INCLUDE_DIRECTORIES "${SwiftCore_INCLUDE_DIR}")
-  find_package_handle_standard_args(SwiftCore DEFAULT_MSG
-    SwiftCore_IMPLIB SwiftCore_INCLUDE_DIR)
+  list(APPEND SwiftCore_NAMES libswiftCore.tbd)
 elseif(LINUX)
-  if (NOT BUILD_SHARED_LIBS)
-    find_path(SwiftCore_INCLUDE_DIR
-      "Swift.swiftmodule"
-      HINTS
-        "${Swift_SDKROOT}/usr/lib/swift_static/linux-static")
-    find_library(SwiftCore_LIBRARY
-      NAMES "libswiftCore.a"
-      HINTS "${Swift_SDKROOT}/usr/lib/swift_static/linux-static")
-    add_library(swiftCore STATIC IMPORTED GLOBAL)
+  if (SwiftCore_USE_STATIC_LIBS)
+    list(APPEND SwiftCore_INCLUDE_DIR_HINTS
+      "${Swift_SDKROOT}/usr/lib/swift_static/linux-static")
+    list(APPEND SwiftCore_LIBRARY_HINTS
+      "${Swift_SDKROOT}/usr/lib/swift_static/linux-static")
+    list(APPEND SwiftCore_NAMES libswiftCore.a)
   else()
-    find_path(SwiftCore_INCLUDE_DIR
-      "Swift.swiftmodule"
-      HINTS
-        "${Swift_SDKROOT}/usr/lib/swift/linux")
-    find_library(SwiftCore_LIBRARY
-      NAMES "libswiftCore.so"
-      HINTS "${Swift_SDKROOT}/usr/lib/swift/linux")
-    add_library(swiftCore SHARED IMPORTED GLOBAL)
+    list(APPEND SwiftCore_INCLUDE_DIR_HINTS
+      "${Swift_SDKROOT}/usr/lib/swift/linux")
+    list(APPEND SwiftCore_LIBRARY_HINTS
+      "${Swift_SDKROOT}/usr/lib/swift/linux")
+    list(APPEND SwiftCore_NAMES libswiftCore.so)
```

Comment:
I think that folding the cases with generator expressions would be reasonable.

Reply:
I would to double check if I can use generator expressions in this context -- my understanding is that we can employ those when setting properties on targets